src/gui/guiFormSpecMenu.cpp

```diff
@@ -4060,6 +4062,34 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 {
 	if (event.EventType==EET_KEY_INPUT_EVENT) {
 		KeyPress kp(event.KeyInput);
+		if (event.KeyInput.PressedDown) {
+			bool ctrl = event.KeyInput.Control;
+			bool shift = event.KeyInput.Shift;
+			if (event.KeyInput.Key == KEY_TAB && ctrl) {
```

Comment:
Please reduce the indent levels to make it more readable. Possible solutions
1. Early `continue` in loops (e.g. by negating the condition)
2. Combining these two `if`s into one
3. Use `goto` to jump to the end of the whole code block
4. Offloading the code to a helper function such that `return` is possible.

Reply:
The goal is to keep the amount of indents in bounds while not sacrificing readability too much. For example, combining two `if`s that check against vastly different variables may make the situation worse.
In this case it's okay, but another low-hanging fruit is `if (s.ftype == f_TabHeader) {`. No other formspec element type would be of interest here. thus you could easily turn into an early `continue` to get rid of an entire level inside the loop.